pkg/snapstore/abs_snapstore.go

```diff
@@ -48,38 +50,45 @@ func NewABSSnapStore(container, prefix, tempDir string, maxParallelChunkUploads
 	if err != nil {
 		return nil, err
 	}
-
 	storageKey, err := GetEnvVarOrError(absStorageKey)
 	if err != nil {
 		return nil, err
 	}
 
-	client, err := storage.NewBasicClient(storageAccount, storageKey)
+	credentials, err := azblob.NewSharedKeyCredential(storageAccount, storageKey)
 	if err != nil {
-		return nil, fmt.Errorf("create ABS client failed: %v", err)
+		return nil, fmt.Errorf("failed to create shared key credentials: %v", err)
 	}
 
-	return GetSnapstoreFromClient(container, prefix, tempDir, maxParallelChunkUploads, &client)
+	p := azblob.NewPipeline(credentials, azblob.PipelineOptions{
+		Retry: azblob.RetryOptions{
+			TryTimeout: downloadTimeout,
+		}})
+	u, err := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", storageAccount))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse service url: %v", err)
+	}
+	serviceURL := azblob.NewServiceURL(*u, p)
+	containerURL := serviceURL.NewContainerURL(container)
+	return GetABSSnapstoreFromClient(container, prefix, tempDir, maxParallelChunkUploads, &containerURL)
 }
 
-// GetSnapstoreFromClient returns a new ABS object for a given container using the supplied storageClient
-func GetSnapstoreFromClient(container, prefix, tempDir string, maxParallelChunkUploads int, storageClient *storage.Client) (*ABSSnapStore, error) {
-	client := storageClient.GetBlobService()
-
+// GetABSSnapstoreFromClient returns a new ABS object for a given container using the supplied storageClient
+func GetABSSnapstoreFromClient(container, prefix, tempDir string, maxParallelChunkUploads int, containerURL *azblob.ContainerURL) (*ABSSnapStore, error) {
```

Comment:
Would it be better to rename this function? Because it now uses `containerURL` instead of a `client` to return the snapstore object.